cmd/copy.go

```diff
@@ -1246,6 +1250,7 @@ func init() {
 	rootCmd.AddCommand(cpCmd)
 
 	// filters change which files get transferred
+	cpCmd.PersistentFlags().BoolVar(&raw.copyContents, "strip-top-dir", false, "copy everything under the source, not the source itself")
```

Comment:
stripTopDir is inferred when wildcarding a local path. Otherwise, yes it is supported with a local source. Because stripTopDir defaults to false, error checking an inconsistency isn't possible because we don't know whether it was _explicitly set_ to false or _defaulted_ to false.

If we wanted to (pointlessly in my opinion) turn it into an enum of (default, false, true), we could manipulate that functionality. However, if a user really wanted the result of --strip-top-dir=false with a wildcard, couldn't they just manually append the top directory to the destination themself?

Reply:
Good point re can't tell why its false.  I'm closing this comment thread.